packages/notte-agent/src/notte_agent/common/types.py

```
@@ -65,6 +66,34 @@
+        actions: list[dict[str, Any]] = []
+        for step in self.trajectory:
+            if only_success and not step.result.success:
+                continue
+            exclude_fields = step.action.non_agent_fields()
+            if id_type == "selector":
+                exclude_fields.remove("selector")
+                exclude_fields.remove("option_selector")
+            exclude_fields.remove("text_label")
+            if "value" in exclude_fields:
+                exclude_fields.remove("value")
+            logger.info(f"Excluding fields: {exclude_fields}")
```

Ruff (0.12.2)

Undefined name `logger`

(F821)